homeassistant/components/fronius/quality_scale.yaml

```diff
@@ -0,0 +1,87 @@
+rules:
```

Comment:
```py

    FroniusSensorEntityDescription(
        key="error_code",
        entity_category=EntityCategory.DIAGNOSTIC,
    ),
    FroniusSensorEntityDescription(
        key="state_code",
        entity_category=EntityCategory.DIAGNOSTIC,
    ),
```

Reply:
State code is the numerical representation returned from the API. `state_message` is the mapped, translated entity. 

For error_code, I'm not sure. API specs say
```
# refer to Ohmpilot manual
# optional field
number CodeOfError 
```

Reply:
Why do we provide both the numerical state as the state message?

for the error code ideally we also show an exhaustive list of messages.

Reply:
The numerical have always been there - as the API responds just a number. 
I've added the translated states later as separate entities to avoid a breaking change.

> for the error code ideally we also show an exhaustive list of messages

That would be nice, but unfortunately I didn't come across a list of those. And I'm not 100% sure if those are consistent across all supported inverter models.